app/controllers/api/tags_controller.rb

```diff
@@ -0,0 +1,48 @@
+class Api::TagsController < Api::TenantController
+  def create
+    @thread = @tenant.message_threads.find(params[:message_thread_id])
+    @tag = @tenant.tags.find_by(name: tag_params)
+    raise ActiveRecord::RecordNotFound unless @tag
+
+    @thread.tags << @tag if @thread.tags.exclude?(@tag)
+  end
+
+  def destroy
+    @thread = @tenant.message_threads.find(params[:message_thread_id])
+    @tag = @tenant.tags.find_by(name: tag_params)
+    @thread.tags.delete(@tag)
+  end
+
+  def batch_add
+    @thread = @tenant.message_threads.find(params[:message_thread_id])
+    @tags = tag_batch_params
+    @result = []
+    @tags.each do |tag_name|
+      tag = @tenant.tags.find_by(name: tag_name)
+      @result << { tag_name => "added" } if tag && @thread.tags.exclude?(tag)
+      @result << { tag_name => "not_found" } unless tag
+      @result << { tag_name => "skipped" } if tag && @thread.tags.include?(tag)
```

Comment:
Toto nutne nepotrebujem v odpovedi. Ale v skutocnosti cakam, ze create a destroy budu robit to co robi batch_add a batch_remove